packages/cardano-services/src/StakePool/TypeormStakePoolProvider/mappers.ts

```diff
@@ -0,0 +1,116 @@
+import { Cardano, Paginated } from '@cardano-sdk/core';
+import { Hash32ByteBase16 } from '@cardano-sdk/crypto';
+import { RelayModel, mapRelay } from '../DbSyncStakePoolProvider';
+
+export type Margin = {
+  numerator: number;
+  denominator: number;
+};
+
+export type PoolModel = {
+  pool_id: string;
```

Comment:
This looks like something that should be handled by typeorm. Is there no way to tell typeorm to map to responses to entity classes when using query builder?

Reply:
If we were fetching the whole entities (`getMany` or `getOne()`) directly with all data in it the response type is matching entity class automatically. However, in our case, we use `getRawMany()` because we want to select some specified fields + some extra SQL computation such as `COUNT(*) OVER()` to get the total count with pagination.
When using raw data I couldn't find a way to say to typeorm to transform raw data fields into entities.
Another disadvantage when working with the query builder and entities is that if fetch the whole entity and over-fetch data we do not need, also at the end we should map again entity type to the Cardano core type as well.

Reply:
Probably worth spending some more time on this to see if we can use typeorm internals to do the mapping for us.
> Another disadvantage when working with the query builder and entities is that if fetch the whole entity

Is it not possible to `SELECT field1, field2`?

Reply:
It's just one way or another based on the needs: https://orkhan.gitbook.io/typeorm/docs/select-query-builder#getting-raw-results
Although there is one opened [issue](https://github.com/typeorm/typeorm/issues/6803) "Mapping raw query to entities" which raises a similar case, it's exposed one solution with the using of [RawSqlResultsToEntityTransformer](https://github.com/typeorm/typeorm/blob/master/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts#L18) but it seems it drastically complicates the "mapping part" compared to the explicit and clear mapper method: "from db model to Cardano core type".